MainModule/Server/Core/Core.luau

```diff
@@ -1223,8 +1223,10 @@ return function(Vargs, GetEnv)
 				local displayName = type(indList) == "table" and table.concat(indList, ".") or tableName
 
 				if type(displayName) == "string" then
-					if ds_blacklist[displayName] then
-						return
+					for i,_ in pairs(ds_blacklist) do
+						if string.match(displayName,`^{i}`) then
```

Comment:
1. When being passed to `string.match()`, `i` should be escaped because it may contain `.` characters.
2. Using a simple pattern match would cause, for example, `IndexA` to be matched but also `IndexAA` even when only `IndexA` is in the blacklist. Though, a quick look through `Core.DS_BLACKLIST` suggests that this shouldn't be a problem in practice.

Allow me to suggest an alternative approach for the sake of correctness and future-proofing:

```luau
for i in ds_blacklist do
	local prefix = string.sub(displayName, 1, #i + 1)
	if string.sub(prefix, 1, #i) == i then
		local delim = string.sub(prefix, #i + 1)
		if delim == "" or delim == "." then
			return
		end
	end
end
```